### PR TITLE
Feature/df 1122  npa domain modelling ac

### PR DESCRIPTION
**Pull request by emina-str**

### CODE REVIEW

File: app-onlineform/src/test/java/ch/admin/seco/alv/service/onlineform/infrastructure/web/controller/ProofOfWorkEffortsRestControllerIntTest.java [hunk 1]
```diff
@@ -153,6 +163,33 @@ public void testAddWorkEffortAsJobSeeker() throws Exception {
 				.andExpect(jsonPath("$.workEfforts.[0].submittedAt").value(equalTo(workEffortDtoToAdd.getSubmittedAt())));
 	}
 
+	@Test
+	@WithJobSeeker
+	public void testAddWorkEffortBeforeValidRangeAsJobSeeker() throws Exception {
```
Comment (pavax) [inline app-onlineform/src/test/java/ch/admin/seco/alv/service/onlineform/infrastructure/web/controller/ProofOfWorkEffortsRestControllerIntTest.java]: rename to testAddWorkEffortBeforeApplyDateValidRange

File: application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsDto.java [hunk 2]
```diff
@@ -44,6 +46,7 @@
 
     private String documentId;
 
+    @Valid
```
Comment (pavax) [inline application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsDto.java]: can be removed

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/ControlPeriod.java [hunk 3]
```diff
@@ -20,6 +20,12 @@
 @Access(AccessType.FIELD)
 public class ControlPeriod {
 
+	public static final int FUTURE_DAYS_FILTER = 5;
```
Comment (pavax) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/ControlPeriod.java]: different constants -> one is to tell the application the submission is always at the fifth and the other has to to with future and the apply-date is a different constant

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/ControlPeriod.java [hunk 4]
```diff
@@ -47,10 +53,14 @@ public YearMonth getValue() {
 
 	public boolean matches(LocalDate date) {
 		final LocalDate endDate = this.endDate();
-		final LocalDate startDate = !isBeforeUnemployment() ? value.atDay(1) : value.minusMonths(3).atDay(1);
+		final LocalDate startDate = this.potentialStartDate();
 		return !(date.isBefore(startDate) || date.isAfter(endDate));
 	}
 
+	public LocalDate potentialStartDate() {
```
Comment (pavax) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/ControlPeriod.java]: you can now also use this method in the DateCalculator

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/ProofOfWorkEfforts.java [hunk 5]
```diff
@@ -221,6 +232,7 @@ public void addWorkEffort(WorkEffort workEffort) {
 		if (this.status == ProofOfWorkEffortsStatus.CLOSED) {
 			throw new BusinessException(PROOF_OF_WORK_EFFORT_CLOSED, "");
 		}
+		// TODO can we add WE in status DRAFT?
```
Comment (pavax) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/ProofOfWorkEfforts.java]: fixme -> do not allow it for now

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/util/ApplyChannelTypeRequiredFields.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = ApplyChannelTypeRequiredFieldsValidator.class)
+public @interface ApplyChannelTypeRequiredFields {
+
+    String message() default "{apply-channel.types.required-fields}";
```
Comment (pavax) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/util/ApplyChannelTypeRequiredFields.java]: if we use jsr-303 we might also need to provide the validation-messages.properties

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/util/PhoneNumber.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+@Size(min = 9,  max = 20, message = "{phone-number.invalid-length.message}")
+@Pattern(regexp = "[+][0-9]{8,}", message = "{phone-number.invalid.message}")
+@Documented
+@Target({ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = {})
+public @interface PhoneNumber {
```
Comment (pavax) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/util/PhoneNumber.java]: inspire yourself by the job-ad-service how the phone-validation was done if we realy need it

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/util/PhoneNumber.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+@Size(min = 9,  max = 20, message = "{phone-number.invalid-length.message}")
+@Pattern(regexp = "[+][0-9]{8,}", message = "{phone-number.invalid.message}")
```
Comment (pavax) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/util/PhoneNumber.java]: make sure that the UI @99999989 send us the phone-number withouth whitespaces (as we did it in the fronent all over where we use phone-numbers)

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/ControlPeriod.java (hunk 3 above)
Comment (emina-str) [reply domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/ControlPeriod.java]: fixed constants.

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/util/PhoneNumber.java (hunk 8 above)
Comment (emina-str) [reply domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/util/PhoneNumber.java]: we use white space - removed PhoneNumber validation from backend.

**Pull request merged**